benchmarks/nel/scripts/datasets/dataset.py

```diff
@@ -73,30 +75,66 @@ def name(self) -> str:
+        model_name (str): Name of model with word vectors to use.
+        n_kb_tokens (int): Number of tokens in entity description to include when inferring embedding.
         """
 
         self._nlp_base = spacy.load(model_name, exclude=["tagger", "lemmatizer"])
+        # self._load_resource("entities")
+        # self._load_resource("failed_entity_lookups")
+        # self._load_resource("annotations")
         print("Parsing external corpus")
-        self._entities, self._failed_entity_lookups, self._annotations = self._parse_external_corpus(**kwargs)
+        self._entities, self._annotations = self._parse_external_corpus()
+        self._entities, self._failed_entity_lookups, self._annotations = \
+            enrich_entities(self._entities, self._annotations)
 
         print(f"Constructing knowledge base with {len(self._entities)} entries")
         self._kb = KnowledgeBase(vocab=self._nlp_base.vocab, entity_vector_length=Dataset.KB_VECTOR_LENGTH)
         entity_list: List[str] = []
         freq_list: List[int] = []
-        vector_list: List[numpy.ndarray] = []
+        vector_list: List[numpy.ndarray] = []  # type: ignore
+        desc_doc_texts: List[str] = []
+        trunc_doc_texts: List[str] = []
+
+        # Prepare and add entities to KB.
         for qid, info in self._entities.items():
             entity_list.append(qid)
-            freq_list.append(info["frequency"])
-            vector_list.append(self._nlp_base(info["description"]).vector)
+            freq_list.append(info.frequency)
+            # todo @RM Use short_description instead of description?
+            desc_doc_texts.append(info.description)
+            trunc_doc_texts.append(" ".join([name.replace("_", " ") for name in info.names]))
+        for i, desc_doc in enumerate(self._nlp_base.pipe(desc_doc_texts, batch_size=100)):
+            trunc_doc_texts[i] += " " + desc_doc[:n_kb_tokens].text
+        for doc in self._nlp_base.pipe(trunc_doc_texts, batch_size=100):
+            vector_list.append(doc.vector if isinstance(doc.vector, numpy.ndarray) else doc.vector.get())
         self._kb.set_entities(entity_list=entity_list, vector_list=vector_list, freq_list=freq_list)
+
+        # Map aliases to their entities. Use entities' page views as priors (this is hacky, since these just reflect
+        # entity popularity, not actual alias-specific priors).
+        aliases_to_entity_ids: Dict[str, Dict[str, int]] = {}
         for qid, info in self._entities.items():
-            for name in info["names"]:
-                self._kb.add_alias(alias=name.replace("_", " "), entities=[qid], probabilities=[1])
+            for alias in info.names:
+                if alias not in aliases_to_entity_ids:
+                    aliases_to_entity_ids[alias] = {}
+                # We could also use e.g. in-corpus frequency instead of page views.
+                aliases_to_entity_ids[alias][qid] = info.pageviews
```

Comment:
Not sure about this, but doesn't Wikipedia have a popularity score that is based on, but different from, pageviews?

Reply:
It looks like the Cirrus Dumps have a ["popularity score"](https://www.mediawiki.org/wiki/Extension:CirrusSearch/Schema#:~:text=production%20search%20clusters-,[popularity_score](https://www.mediawiki.org/wiki/Extension:CirrusSearch/Schema#:~:text=production%20search%20clusters-,popularity_score)), which is what I was remembering. It is also pageview based so I am not sure it is an improvement, and it sounds like it isn't available in other sources.

Reply:
Since we are switching to Wiki dumps and the existing scripts already parse out alias-entity priors, I'd forego pageviews, popularity and other measures for now. It might very well make sense to incorporate those later, but for now I'd stick with the [approach based on interwiki-links available in the scripts](https://github.com/explosion/projects/blob/8ab0b3b873f6c2393e9f1d9b9b30238ef1dddd8a/nel-wikipedia/wikipedia_processor.py#L59).